### DISCUSSION TITLE
investigating Neewer ZC-10S Lighting panel set

### DISCUSSION THREAD

**Discussion by csylvain** [merbanan/rtl_433#2498] [Q&A]
I have a pair of Neewer ZC-10S LED light panels - each panel has its own three button 433.920 MHz remote. one panel is coded "red" and the other "white". I have two aims: 1. to add this product to rtl_433, and 2. be able to emit the remote signals with a PlutoSDR. This is my first foray into using rtl_433 to investigate a RF remote controlled product.

### What I've Done So Far

I am using `rtl_433-rtlsdr-soapysdr-tls.exe` from `rtl_433-win-msvc-x64-nightly` (d/l April 30) with a PlutoSDR. In order to get that working with `PothosSDR-2021.07.25-vc16-x64`, it was necessary to rename all the DLL files in the nightly folder to *.DLO and then run rtl_433 from the PothosSDR bin folder.

#### Analysis with *-A*
```
https://triq.org/pdv/#AAB103019804D82F848190819081819081819081909090908181818181818190818255

Pulses	16× 408.0 ±0.0 µs	9× 1240.0 ±0.0 µs
Gaps	9× 408.0 ±0.0 µs	15× 1240.0 ±0.0 µs
Periods	24× 1648.0 ±0.0 µs
Timings	25× 408.0 ±0.0 µs	24× 1240.0 ±0.0 µs	1× 12164.0 ±0.0 µs
DC bias (Pulse/Gap skew): -20.6%
Guessing modulation: Pulse Width Modulation with fixed period
modulation: PWM short: 408.0 long: 1240.0 sync: - gap: - reset: 1488.0
RfRaw (rx): AAB103019804D82F848190819081819081819081909090908181818181818190818255
RfRaw (tx): AAB0210301019804D82F848190819081819081819081909090908181818181818190818255
Bits: {25} AD A1 FD 8
```
#### Many Experiments Later
I think I have correctly narrowed down the configuration to
```
rtl_433-rtlsdr-soapysdr-tls.exe -v -d "driver=plutosdr" \
  -R 0 -X 'n=try3,m=OOK_PWM,s=388,l=1204,r=1280,g=0,t=326,y=0,bits>=25,match={17}6f4af' \
  -X 'n=try4,m=OOK_PWM,s=388,l=1204,r=1280,g=0,t=326,y=0,bits>=25,match={17}ada1f'
```
which shows all 6 possible key code transmissions from "red" (0xada1f) and "white" (0x6f4af) with no junk decodes.
The one nibble before the trailing 8 is the unique value for each keypress.
### What Next
* I tried to find examples of using rtl_433 to produce a recording of the detected transmissions. Is this easily done?
* I am not sure how to give names to the key codes. Is this where creating a conf file completes the job?
* Am I missing something? Is there more to do to settle on being able to decode and display the transmissions?

**Comment by zuckschwerdt** [comment]
Yes, removing the DLLs and running from the Pothos folder is the right way to get the Pothos libs to load.

Looks like a x1527 protocol, 24 OOK PWM bits plus 1 sync. The x1527 encoders allow for the last bits to encode some state, like which button is pressed. If those are the last 4 bits then just e.g. `match=ada1f` should do, it's short for `match={20}ada1f`.

Omit sync (`y`) and gap (`g`) and make reset a little wider, say `r=1400`.

Record with `-S known` if you already have a working `-X`. You should be able to play back with https://github.com/triq-org/tx_tools/

Look at https://github.com/merbanan/rtl_433/tree/master/conf for some other remotes with button codes.
